### PR TITLE
Java French DateTime Implementation

### PR DESCRIPTION
**Pull request by mrussek**
This implements French DateTime parsing in Java, largely by filling in from the C# version where necessary. It appears to be passing all of the test cases that are passing on master, with the sole exception of `"Je vais partir pour 3.5ans"` which I'm not sure _should_ be passing, since I think that the French number recognizer expects a format of "3,5" for the fractional part. Additionally, it appears that some `OrdinalModel` and `NumberModel tests are failing, in particular:
```
  test[NumberRecognizer - French - NumberModel - "sous la pluie et un vent glacial"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match text on Input: "sous la pluie et un vent glacial" expected:<[]un> but was:<[et ]un>
  test[NumberRecognizer - French - OrdinalModel - "Son dizième anniversaire"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match "Result Count" on Input: "Son dizième anniversaire" expected:<1> but was:<0>
  test[NumberRecognizer - French - OrdinalModel - "1ère femme"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match "Result Count" on Input: "1ère femme" expected:<1> but was:<0>
  test[NumberRecognizer - French - OrdinalModel - "quinzième, dix-huirième, dix-septième, septième"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match "Result Count" on Input: "quinzième, dix-huirième, dix-septième, septième" expected:<4> but was:<2>
  test[NumberRecognizer - French - OrdinalModel - "Le vingt et unième jour"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match resolution.value on Input: "Le vingt et unième jour" expected:<2[1]> but was:<2[0]>
```
but these all appear to be failing on the latest master as well.

Additionally, I need to address some checkstyle issues, but while I do that, I would like any and all feedback on the code. Thank you!

### CODE REVIEW

File: Specs/DateTime/French/DateExtractor.json [hunk 1]
```diff
@@ -947,7 +947,7 @@
   },
   {
     "Input": "Qui ai-je envoye un mois il y a",
-    "NotSupported": "javascript, dotnet, python",
+    "NotSupported": "javascript, dotnet, python, java",
```
Comment (mrussek) [inline Specs/DateTime/French/DateExtractor.json:950]: I assumed that since this was not supported by any of the other languages, it would not be supported for Java either.

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseSetExtractor.java [hunk 2]
```diff
@@ -138,9 +138,9 @@ public BaseSetExtractor(ISetExtractorConfiguration config) {
         matches = RegExpUtility.getMatches(setWeekDayRegex, input);
         for (Match match : matches) {
             if (match != null) {
-                String trimedText = sb.delete(match.index, match.index + match.length).toString();
```
Comment (mrussek) [inline Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseSetExtractor.java:141]: This seemed to be causing failures in the course of french tests. I switched to not using the "mutable" string builder which is modified between different loop iterations.

File: Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/french/parsers/FrenchNumberParserConfiguration.java [hunk 3]
```diff
@@ -20,7 +20,7 @@ public FrenchNumberParserConfiguration() {
     }
 
     public FrenchNumberParserConfiguration(NumberOptions options) {
-        this(new CultureInfo(Culture.Portuguese), options);
+        this(new CultureInfo(Culture.French), options);
```
Comment (mrussek) [inline Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/french/parsers/FrenchNumberParserConfiguration.java:23]: Despite this not being related to datetimes, I figured this is more accurate?

File: Specs/DateTime/French/SetParser.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   {
     "Input": "Je vais partir a 9am Dimanches",
     "Context": {
-      "ReferenceDateTime": "2017-10-09T12:21:16.6159645-03:00"
+      "ReferenceDateTime": "2017-10-09T12:21:16.6159645+03:00"
```
Comment (mrussek) [inline Specs/DateTime/French/SetParser.json:5]: The parser for `"ReferenceDateTime"` seemed to be assuming that the timezone delimiter was `+` and not `-`.

File: Specs/DateTime/French/DateExtractor.json (hunk 1 above)
Comment (tellarin) [reply Specs/DateTime/French/DateExtractor.json:950]: Yes, it should be OK to temporarily mark it as NotSupported until the rest of the support is ported.

File: Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/french/parsers/FrenchNumberParserConfiguration.java (hunk 3 above)
Comment (tellarin) [reply Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/french/parsers/FrenchNumberParserConfiguration.java:23]: Thanks for the catch.

File: Specs/DateTime/French/SetParser.json (hunk 4 above)
Comment (tellarin) [reply Specs/DateTime/French/SetParser.json:5]: Both should work. As timezone offsets can be either positive or negative, it would be better to track the root cause of the issue than to make this change in the spec.

**Comment by tellarin** [comment]
@mrussek, thanks a lot for the PR. We'll review soon. Meanwhile, I've added some comments here. 
Regarding the issues you mention in the PR description, where do you see them happening? All tests pass in the CI build for /master.

File: Specs/DateTime/French/DateTimeModel.json [hunk 5]
```diff
@@ -95,13 +95,13 @@
     ]
   },
   {
-    "Input": "Je vais partir pour 3.5ans",
+    "Input": "Je vais partir pour 3,5ans",
```
Comment (tellarin) [inline Specs/DateTime/French/DateTimeModel.json:98]: Please don't make such edits in specs. Previously existing cases that pass must continue to pass.
In some variants of French the decimal separator can also be ".".

File: Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java [hunk 6]
```diff
@@ -247,6 +260,35 @@ private static IDateTimeParser getSpanishParser(String name) {
...
+                return new BaseDateTimeParser(new FrenchDateTimeParserConfiguration(new FrenchCommonDateTimeParserConfiguration(DateTimeOptions.None)));
+            case "DateTimePeriodParser":
+                return new BaseDateTimePeriodParser(new FrenchDateTimePeriodParserConfiguration(new FrenchCommonDateTimeParserConfiguration(DateTimeOptions.None)));
+            case "DurationParser":
+                return new BaseDurationParser(new FrenchDurationParserConfiguration(new FrenchCommonDateTimeParserConfiguration(DateTimeOptions.None)));
+            case "HolidayParser":
+                return new BaseHolidayParser(new FrenchHolidayParserConfiguration());
+            case "SetParser":
+                return new BaseSetParser(new FrenchSetParserConfiguration(new FrenchCommonDateTimeParserConfiguration(DateTimeOptions.None)));
+            //case "MergedParser":
```
Comment (tellarin) [inline Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java:282]: Why is this disabled?

File: Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java [hunk 7]
```diff
@@ -247,6 +260,35 @@ private static IDateTimeParser getSpanishParser(String name) {
...
         }
     }
+    private static IDateTimeParser getFrenchParser(String name) {
+
+        switch (name) {
+            case "DateParser":
+                return new BaseDateParser(new FrenchDateParserConfiguration(new FrenchCommonDateTimeParserConfiguration(DateTimeOptions.None)));
+            case "DatePeriodParser":
+                return new BaseDatePeriodParser(new FrenchDatePeriodParserConfiguration(new FrenchCommonDateTimeParserConfiguration(DateTimeOptions.None)));
+            //case "DateTimeAltParser":
```
Comment (tellarin) [inline Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java:270]: This is OK, as AlternativeTypes currently only exists for English and requires special flags to enable it.

File: Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeExtractorTest.java [hunk 8]
```diff
@@ -108,14 +122,17 @@ public static IDateTimeExtractor getExtractor(String language, String modelName)
...
+            case Culture.Spanish:
+                return getSpanishExtractor(modelName);
+            case Culture.French:
+                return getFrenchExtractor(modelName);
+            default:
+                throw new AssumptionViolatedException("Extractor Type/Name not supported.");
             }
-        } catch (IllegalArgumentException ex) {
+        }
+        catch (IllegalArgumentException ex) {
```
Comment (tellarin) [inline Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeExtractorTest.java:135]: Nitpick: This should be in the above line.

File: Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeExtractorTest.java [hunk 9]
```diff
@@ -124,71 +141,102 @@ private static IDateTimeExtractor getEnglishExtractor(String name) {
...
+        case "DateTimePeriodExtractor":
+            return new BaseDateTimePeriodExtractor(new EnglishDateTimePeriodExtractorConfiguration());
+        case "DurationExtractor":
+            return new BaseDurationExtractor(new EnglishDurationExtractorConfiguration());
+        case "HolidayExtractor":
+            return new BaseHolidayExtractor(new EnglishHolidayExtractorConfiguration());
+        case "MergedExtractor":
+            return new BaseMergedDateTimeExtractor(new EnglishMergedExtractorConfiguration(DateTimeOptions.None));
+        case "MergedExtractorSkipFromTo":
+            return new BaseMergedDateTimeExtractor(
```
Comment (tellarin) [inline Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeExtractorTest.java:161]: Nitpick: Please don't change previous formatting unless an error was being raised. Can you revert?

File: Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeExtractorTest.java [hunk 10]
```diff
@@ -124,71 +141,102 @@ private static IDateTimeExtractor getEnglishExtractor(String name) {
 
         IOptionsConfiguration config = new BaseOptionsConfiguration();
         switch (name) {
-            case "DateExtractor":
```
Comment (tellarin) [inline Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeExtractorTest.java:127]: These changes were only line break changes? Please be careful to avoid making unnecessary changes unless it was for consistency with the rest of the file. Such changes can lead to less usability in code review tools and more effort required in the reviews.

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseSetExtractor.java (hunk 2 above)
Comment (tellarin) [reply Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseSetExtractor.java:141]: Can you mention which tests failed? It would be strange that other languages don't fail and now only French would.

**Comment by mrussek** [comment]
Hi @tellarin, some updates.

Since I did this work on the part of my employer I'm having to get them to approve the CLA associated with the contribution.

Sorry about the formatting changes. I did the work using my custom formatter in IntelliJ, and only after completing imported the checkstyle rules and reformatted. It seems that the checkstyle reformat was incomplete. My mistake for not checking.

The failures that I mentioned on master seem to be still occurring on a fresh clone of the repo, pointed at the master branch, when executing `mvn test` in the `Java` directory. This is the full list of failures on master that I observe:
```
Failed tests:   test[NumberWithUnitRecognizer - English - DimensionModel - "The submarine mein weigh 25 lakh tonnes"](com.microsoft.recognizers.text.tests.numberwithunit.NumberWithUnitTest): Does not match text on Input: "The submarine mein weigh 25 lakh tonnes" expected:<[25 lakh ]tonnes> but was:<[]tonnes>
  test[NumberWithUnitRecognizer - English - CurrencyModel - "Am I eligible for a lone of 35 lakh rupees"](com.microsoft.recognizers.text.tests.numberwithunit.NumberWithUnitTest): Does not match text on Input: "Am I eligible for a lone of 35 lakh rupees" expected:<[35 lakh ]rupees> but was:<[]rupees>
  test[NumberRecognizer - English - NumberModel - "I can give you 13 lakh"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match text on Input: "I can give you 13 lakh" expected:<13[ lakh]> but was:<13[]>
  test[NumberRecognizer - English - NumberModel - "There are 1 crore boxes in this container"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match text on Input: "There are 1 crore boxes in this container" expected:<1[ crore]> but was:<1[]>
  test[NumberRecognizer - English - NumberModel - "The total is three lakh crore"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match text on Input: "The total is three lakh crore" expected:<three[ lakh crore]> but was:<three[]>
  test[NumberRecognizer - French - NumberModel - "sous la pluie et un vent glacial"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match text on Input: "sous la pluie et un vent glacial" expected:<[]un> but was:<[et ]un>
  test[NumberRecognizer - French - OrdinalModel - "Son dizième anniversaire"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match "Result Count" on Input: "Son dizième anniversaire" expected:<1> but was:<0>
  test[NumberRecognizer - French - OrdinalModel - "1ère femme"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match "Result Count" on Input: "1ère femme" expected:<1> but was:<0>
  test[NumberRecognizer - French - OrdinalModel - "quinzième, dix-huirième, dix-septième, septième"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match "Result Count" on Input: "quinzième, dix-huirième, dix-septième, septième" expected:<4> but was:<2>
  test[NumberRecognizer - French - OrdinalModel - "Le vingt et unième jour"](com.microsoft.recognizers.text.tests.number.NumberTest): Does not match resolution.value on Input: "Le vingt et unième jour" expected:<2[1]> but was:<2[0]>
```
One possible discrepancy between my machine and the CI environment is the JDK version (for me, 8.0.242 of the Zulu JDK) and the version of maven (there is no repository version of maven such as a `mvnw` script, my system version is 3.6.0). It seems these failures on my system are related to new specs being added for English and French. I will try again shortly with other JDK versions.

I have some questions regarding the spec files that I will be asking inline on the PR.

Thanks again for taking a look!

File: Specs/DateTime/French/DateTimeModel.json (hunk 5 above)
Comment (mrussek) [reply Specs/DateTime/French/DateTimeModel.json:98]: I can change it back, but it seems that the number parser will need to support this as well. Currently the number parser (used by the date parser on the substring "3.5" in the above case), returns "35" as the value for parsing "3.5".

Do you know if it is acceptable in French to use "." as a decimal separator _only_ within dates? Or with years specifically? I haven't had a chance to look at how the code for other programming languages handles this case. I will do this now.

Part of what motivated me to change it was that I noticed other test cases use a comma decimal separator. Notably `"je partirai pour 3,5 seconde "` on line 223 in DurationParser.json.

File: Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java (hunk 6 above)
Comment (mrussek) [reply Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java:282]: Ah, my bad. I copied the switch for spanish in this code, where this is also disabled. I will enabled it here.

File: Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java (hunk 7 above)
Comment (mrussek) [reply Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java:270]: This is interesting. If I enable it, all tests still remain passing. Should I do so?

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseSetExtractor.java (hunk 2 above)
Comment (mrussek) [reply Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseSetExtractor.java:141]: These are the tests which fail:

1. "Je vais partir 9am chaque lundis"
```
test[DateTimeRecognizer - French - SetExtractor - "Je vais partir 9am chaque lundis"](com.microsoft.recognizers.text.tests.datetime.DateTimeExtractorTest)  Time elapsed: 0.027 sec  <<< ERROR!
java.lang.StringIndexOutOfBoundsException
	at java.lang.AbstractStringBuilder.delete(AbstractStringBuilder.java:760)
	at java.lang.StringBuilder.delete(StringBuilder.java:244)
	at com.microsoft.recognizers.text.datetime.extractors.BaseSetExtractor.matchEach(BaseSetExtractor.java:141)
	at com.microsoft.recognizers.text.datetime.extractors.BaseSetExtractor.extract(BaseSetExtractor.java:42)
	at com.microsoft.recognizers.text.tests.datetime.DateTimeExtractorTest.extract(DateTimeExtractorTest.java:113)
	at com.microsoft.recognizers.text.tests.datetime.DateTimeExtractorTest.recognizeAndAssert(DateTimeExtractorTest.java:86)
	at com.microsoft.recognizers.text.tests.AbstractTest.test(AbstractTest.java:115)
```
2. "Je vais partir tous les matins"
```
test[DateTimeRecognizer - French - SetExtractor - "Je vais partir tous les matins"](com.microsoft.recognizers.text.tests.datetime.DateTimeExtractorTest)  Time elapsed: 0.028 sec  <<< ERROR!
java.lang.StringIndexOutOfBoundsException
	at java.lang.AbstractStringBuilder.delete(AbstractStringBuilder.java:760)
	at java.lang.StringBuilder.delete(StringBuilder.java:244)
	at com.microsoft.recognizers.text.datetime.extractors.BaseSetExtractor.matchEach(BaseSetExtractor.java:141)
	at com.microsoft.recognizers.text.datetime.extractors.BaseSetExtractor.extract(BaseSetExtractor.java:42)
	at com.microsoft.recognizers.text.tests.datetime.DateTimeExtractorTest.extract(DateTimeExtractorTest.java:113)
	at com.microsoft.recognizers.text.tests.datetime.DateTimeExtractorTest.recognizeAndAssert(DateTimeExtractorTest.java:86)
	at com.microsoft.recognizers.text.tests.AbstractTest.test(AbstractTest.java:115)
```

It could be an error in the French implementation, but I'm not convinced that the original implementation of this method matches what, say, its Python counterpart is doing.

```python
    for match in regex.finditer(self.config.set_each_regex, source):
            trimmed_source = source[0:match.start()] + source[match.end():]

            for extract_result in extractor.extract(trimmed_source, reference):
                if extract_result.start <= match.start() < extract_result.start + extract_result.length:
                    yield Token(extract_result.start, extract_result.start + extract_result.length + len(match.group()))

        for match in regex.finditer(self.config.set_week_day_regex, source):
            trimmed_source = source[0:match.start()] \
                + RegExpUtility.get_group(match, Constants.WEEKDAY_GROUP_NAME) + source[match.end():]

            for extract_result in extractor.extract(trimmed_source, reference):
                if extract_result.start <= match.start() and\
                        RegExpUtility.get_group(match, Constants.WEEKDAY_GROUP_NAME) in extract_result.text:
                    length = extract_result.length + 1
                    prefix = RegExpUtility.get_group(match, Constants.PREFIX_GROUP_NAME)
                    if prefix:
                        length += len(prefix)

                    yield Token(extract_result.start, extract_result.start + length)
```
Note that each pass through the for loop, the Python version re-extracts the substring from the (unmutated) input. This is different than the original Java version, which reassigns `sb` to a new `StringBuilder` with the _trimmed_ text for the next iteration of the loop. This results in the deletion of a range of indices from the string builder which **does not** correspond to the text that was matched.

File: Specs/DateTime/French/SetParser.json (hunk 4 above)
Comment (mrussek) [reply Specs/DateTime/French/SetParser.json:5]: I will try and fix the root cause issue.

File: Specs/DateTime/French/DateTimeModel.json (hunk 5 above)
Comment (mrussek) [reply Specs/DateTime/French/DateTimeModel.json:98]: I think I figured this out. There was a "mistranslation" into Java code in `BaseNumberParser`. More details at that location.

File: Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/parsers/BaseNumberParser.java [hunk 11]
```diff
@@ -419,9 +419,9 @@ private boolean skipNonDecimalSeparator(char ch, int distance, CultureInfo cultu
         // Special cases for multi-language countries where decimal separators can be used interchangeably. Mostly informally.
         // Ex: South Africa, Namibia; Puerto Rico in ES; or in Canada for EN and FR.
         // "me pidio $5.00 prestados" and "me pidio $5,00 prestados" -> currency $5
-        Pattern cultureRegex = Pattern.compile("^(en|es|fr)(-)?\b", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CHARACTER_CLASS);
+        Pattern cultureRegex = Pattern.compile("^(en|es|fr)(-)?\\b", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CHARACTER_CLASS);
```
Comment (mrussek) [inline Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/parsers/BaseNumberParser.java:422]: There were two issues preventing the appropriate parsing of "3.5ans".
1. The first is that in this regex, "\b" was used, but in Java this is an escape code for a backspace, and does not correspond to a word boundary.
2. The `matches` method was used even though it seems only a substring match was desired. `matches` requires the whole string to match. Changing this to `find` returns true for "fr-fr" and causes the whole return expression to evaluate to false.

Comment (tellarin) [reply Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/parsers/BaseNumberParser.java:422]: Thanks for catching this. Can you add a `@TODO` comment here calling attention to refactor it?
This language list should not be hardcoded there. Ideally it should come from a project config-level file so all platforms are always in sync.

**Comment by tellarin** [comment]
@mrussek, BTW, whenever we can merge this, we'll propagate your assert changes to the other platforms. They make finding the error cases much easier. 
Why  weren't them like that from the beginning, huh? 😃

**Comment by mrussek** [comment]
@tellarin , oops!  I removed those changes in my last update. I can add them back though.

**Comment by mrussek** [comment]
@tellarin, it looks like some specs were added for French in `MergedParser.json` that have incorrect `values` fields:
```json
{                                                  
  "Input": "organiser une réunion dans 3 semaines",
  "Context": {                                     
    "ReferenceDateTime": "2016-11-07T00:00:00"     
  },                                               
  "NotSupported": "javascript",                    
  "Results": [                                     
    {                                              
      "Text": "dans 3 semaines",                   
      "Type": "datetimeV2.date",                   
      "Value": {                                   
        "values": []                               
      },                                           
      "Start": 22,                                 
      "Length": 15                                 
    }                                              
  ]                                                
},                                                 
{                                                  
  "Input": "organiser une réunion dans 3 mois",    
  "Context": {                                     
    "ReferenceDateTime": "2016-11-07T00:00:00"     
  },                                               
  "NotSupported": "javascript",                    
  "Results": [                                     
    {                                              
      "Text": "dans 3 mois",                       
      "Type": "datetimeV2.date",                   
      "Value": {                                   
        "values": []                               
      },                                           
      "Start": 22,                                 
      "Length": 11                                 
    }                                              
  ]                                                
},                                                 
{                                                  
  "Input": "organiser une réunion dans 3 annees",  
  "Context": {                                     
    "ReferenceDateTime": "2016-11-07T00:00:00"     
  },                                               
  "NotSupported": "javascript",                    
  "Results": [                                     
    {                                              
      "Text": "dans 3 annees",                     
      "Type": "datetimeV2.date",                   
      "Value": {                                   
        "values": []                               
      },                                           
      "Start": 22,                                 
      "Length": 13                                 
    }                                              
  ]                                                
},                                                                                   
```
All of these test cases have 0 values listed, even though they have values. This is causing the Java implementation to fail because it finds the (correct) values for these phrases.

**Comment by mrussek** [comment]
@tellarin Any idea on those three utterances? For what it's worth, here are the results of the Java and Python implementations respectively.

Java:
```
Utterance: "organiser une réunion dans 3 semaines"
Reference: 2016-11-07T00:00
Resolution: {values=[{timex=2016-11-28, type=date, value=2016-11-28}]}

Utterance: "organiser une réunion dans 3 mois"
Reference: 2016-11-07T00:00
Resolution: {values=[{timex=2017-02-07, type=date, value=2017-02-07}]}

Utterance: "organiser une réunion dans 3 annees"
Reference: 2016-11-07T00:00
Resolution: {values=[{timex=2019-11-07, type=date, value=2019-11-07}]}
```
Python:
```
Utterance: "organiser une réunion dans 3 semaines"
Reference: 2016-11-07 00:00:00
Resolution: {'values': [{'timex': '(2016-11-22,2016-11-29,P1W)', 'type': 'daterange', 'start': '2016-11-22', 'end': '2016-11-29'}]}

Utterance: "organiser une réunion dans 3 mois"
Reference: 2016-11-07 00:00:00
Resolution: {'values': [{'timex': '(2017-01-08,2017-02-08,P1M)', 'type': 'daterange', 'start': '2017-01-08', 'end': '2017-02-08'}]}

Utterance: "organiser une réunion dans 3 annees"
Reference: 2016-11-07 00:00:00
Resolution: {'values': [{'timex': '(2018-11-08,2019-11-08,P1Y)', 'type': 'daterange', 'start': '2018-11-08', 'end': '2019-11-08'}]}
```

I'm not a native French speaker, but the Java results seem more sensible? Since "in 3 months" etc seems to imply a single date, rather than a daterange? Additionally, neither of these match the Spec for these cases, with no values whatsoever. I'm currently debugging to see why the Python tests still pass.

**Comment by tellarin** [comment]
Sorry for the delay to reply, @mrussek. we'll try to take a look and get back to you soon.

**Comment by mrussek** [comment]
@tellarin no worries. Just got confirmation from my employer that I can sign the CLA!

**Comment by tellarin** [comment]
The Java results are indeed more correct according to the current design. And they seem to match the .NET ones at the model level.
It seems there may be something wrong with the test project in .NET too that's not properly validating these cases.

**Comment by tellarin** [comment]
We've found the test issue and fixed the relevant specs. Your PR should be able to merge once the fix is in.
